packages/material-ui/src/Select/SelectInput.js

```diff
@@ -305,6 +281,8 @@ const SelectInput = React.forwardRef(function SelectInput(props, ref) {
         anchorEl={displayRef.current}
         open={open}
         onClose={handleClose}
+        disableRestoreFocus
```

Comment:
The trigger isn't focused when opening (to prevent a label shrink flash) so restore won't do anything.